homeassistant/components/isy994.py

```diff
@@ -0,0 +1,209 @@
+"""
+Connects to an ISY-994 controller and loads relevant components to control its
+devices. Also contains the base classes for ISY Sensors, Lights, and Switches.
+"""
+# system imports
+import logging
+from urllib.parse import urlparse
+
+# addon library imports
+import PyISY
+
+# homeassistant imports
+from homeassistant import bootstrap
+from homeassistant.loader import get_component
+from homeassistant.helpers import validate_config
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.const import (
+    CONF_HOST, CONF_USERNAME, CONF_PASSWORD, EVENT_PLATFORM_DISCOVERED,
+    ATTR_SERVICE, ATTR_DISCOVERED, ATTR_FRIENDLY_NAME)
+
+# homeassistant constants
+DOMAIN = "isy994"
+DEPENDENCIES = []
+DISCOVER_LIGHTS = "isy994.lights"
+DISCOVER_SWITCHES = "isy994.switches"
+DISCOVER_SENSORS = "isy994.sensors"
+ISY = None
+SENSOR_STRING = 'Sensor'
+HIDDEN_STRING = '{HIDE ME}'
+
+# setup logger
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """
+    Setup isy994 component.
+    This will automatically import associated lights, switches, and sensors.
+    """
+    # pylint: disable=global-statement
+    # check for required values in configuration file
+    if not validate_config(config,
+                           {DOMAIN: [CONF_HOST, CONF_USERNAME, CONF_PASSWORD]},
+                           _LOGGER):
+        return False
+
+    # pull and parse standard configuration
+    user = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    host = urlparse(config[DOMAIN][CONF_HOST])
+    addr = host.geturl()
+    if host.scheme == 'http':
+        addr = addr.replace('http://', '')
+        https = False
+    elif host.scheme == 'https':
+        addr = addr.replace('https://', '')
+        https = True
+    else:
+        _LOGGER.error('isy994 host value in configuration file is invalid.')
+        return False
+    port = host.port
+    addr = addr.replace(':{}'.format(port), '')
+
+    # pull and parse optional configuration
+    global SENSOR_STRING
+    global HIDDEN_STRING
+    SENSOR_STRING = str(config[DOMAIN].get('sensor_string', SENSOR_STRING))
+    HIDDEN_STRING = str(config[DOMAIN].get('hidden_string', HIDDEN_STRING))
+
+    # connect to ISY controller
+    global ISY
+    ISY = PyISY.ISY(addr, port, user, password, use_https=https, log=_LOGGER)
+    if not ISY.connected:
+        return False
+
+    # Load components for the devices in the ISY controller that we support
+    for comp_name, discovery in ((('sensor', DISCOVER_SENSORS),
```

Comment:
It would be nice if we could for example only load and setup a sensor component if the ISY controller actually contains a sensor component.